DeeployTest/testUtils/core/execution.py

```
@@ -27,31 +74,175 @@ def generate_network(config: DeeployTestConfig, skip: bool = False) -> None:
+            config.n_accum_steps = meta["n_accum_steps"]
+            config.training_num_data_inputs = meta["training_num_data_inputs"]
+            log.info(f"[Execution] Training meta: {meta}")
+
+        # --- Step 2: Tiled optimizer network (SGD via testMVPOptimizer.py) ---
+        opt_dir = _resolve_optimizer_dir(config)
+        opt_script = script_dir / "testMVPOptimizer.py"
+
+        if not Path(opt_dir).exists():
+            log.warning(f"Optimizer directory not found: {opt_dir} — skipping optimizer codegen")
+        elif not opt_script.exists():
+            log.warning(f"testMVPOptimizer.py not found — skipping optimizer codegen")
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)